### ISSUE TITLE
Pretrained Weight Sharing 

### ISSUE THREAD

**Issue by terenceylchow124** [orionw/RedditHumorDetection#3]
Hi Orion, It's the interesting paper in humor detection that using the Transformer-based model back in 2019 while reddit dataset have been built. I'm quite new in this field, and recurrent conducting the project about humor detection also. I've been successfully trained the provided model. However, due to the hardware setup, I am not able to train the model with batch-size=256 (I can almost use 32 as bs in bert base model. Therefore, It would be more than happy if you would share your trained weight also. Thanks so much.

Terence

**Comment by orionw** [comment]
Hi Terence!  Thank you for your interest in this project!

Unfortunately, Github is not the best place to store large model weights, so I wasn't able to store them here.  I don't think I will be able to provide the model weights either - it's been several years since I last used this repository and I no longer have the model weights stored anywhere.

However, there are a couple things you can do, if you want to reproduce this: (1) you could run the BERT model with a smaller batch size of 16 or 32.  This may not give you the exact same accuracy that the original script would, but it should be relatively close.  (2) If those results turn out to not be good enough, you can modify the `gradient_accumulation_steps` to simulate a higher batch size.  If you change the batch size to 16 from 256 (a factor of 16) you can then change the gradient accumulation steps to increase by that factor (so from the current amount of 24 to 24*16) to counteract it.  This should produce the same results as using a larger batch size (albeit, the training might take longer).  For more information about gradient accumulation, see articles like [this one](https://towardsdatascience.com/gradient-accumulation-overcoming-memory-constraints-in-deep-learning-36d411252d01). 

Best of luck to you in your research!
